### PR TITLE
Added permessage defalte comperssion

### PR DESCRIPTION
**Pull request by jszczygiel**
Changed okhttp websocket implementation to nv-websocket-client, which is more mature and supports permessage defalte compresion

### CODE REVIEW

**Comment by nkzawa** [comment]
Thanks for your PR, but let me check about `nv-websocket-client`. Actually I'm not sure which is better.

File: pom.xml [hunk 1]
```diff
@@ -1,220 +1,235 @@
...
-  <artifactId>engine.io-client</artifactId>
-  <version>0.7.1-SNAPSHOT</version>
-  <packaging>jar</packaging>
-  <name>engine.io-client</name>
-  <description>Engine.IO Client Library for Java</description>
-  <url>https://github.com/socketio/engine.io-client-java</url>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
```
Comment (nkzawa) [inline pom.xml]: Any reasons of style change? Can you create another PR for it if possible ?

File: src/main/java/io/socket/engineio/client/transports/WebSocket.java [hunk 2]
```diff
@@ -47,101 +37,93 @@ protected void doOpen() {
...
+                public void onError(com.neovisionaries.ws.client.WebSocket websocket, final WebSocketException cause) throws Exception {
+                    EventThread.exec(new Runnable() {
+                        @Override
+                        public void run() {
+                            self.onError("websocket error", cause);
+                        }
+                    });
+                }
+            });
+            ws.connect();
```
Comment (nkzawa) [inline src/main/java/io/socket/engineio/client/transports/WebSocket.java]: It has to be `ws.connectAsynchronously()` ?

File: pom.xml (hunk 1 above)
Comment (jszczygiel) [inline pom.xml]: Yep, sorry about that. Havent noticed that my formatter changed style.

**Comment by nkzawa** [comment]
With https://github.com/socketio/engine.io-client-java/pull/85, you can implement your own websocket interface and should allow to add deflate to websocket.

**Pull request closed without merging**